buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -412,6 +413,31 @@ class ClusterFormationTasks {
         return parentTask
     }
 
+    /** Configure ES JAVA OPTS - adds system properties, assertion flags, remote debug etc */
+    static Task configureESJavaOpts(String parent, Project project, Task setup, NodeInfo node) {
```

Comment:
I don't think we need a separate task for it, especially that the result is only stored in memory. 
I would add this to the `doFirst` we already run on the `start` task.
We can just set ES_JAVA_OPTS before the  start task, or concatenate it if already set.
The builds script could also set this. None of the currently do, but we should either concatenate to it if set, or fail if it's set if we don;t want to support it.  I'm not saying we shouldn't just speculating on what the initial intent could have been.

Reply:
@atorok do you mean to put it to https://github.com/elastic/elasticsearch/blob/db90905405297a392707c3137e19f7227462bdd8/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy#L605 ?

Reply:
exactly